test/xbgpu/test_engine.py

```diff
@@ -1046,28 +1079,41 @@ def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.Heap
 
         # Also need to access the request arguments later when generating expected sensor updates
         rng = np.random.default_rng(seed=1)
-        weights = rng.uniform(0.5, 2.0, size=(len(beam_outputs), n_ants))
-        quant_gains = rng.uniform(0.5, 2.0, size=(len(beam_outputs)))
-        delays = np.zeros((len(beam_outputs), n_ants, 2), np.float64)
+        weights = rng.uniform(0.5, 2.0, size=(N_BEAM_REQUESTS, len(beam_outputs), n_ants))
+        quant_gains = rng.uniform(0.5, 2.0, size=(N_BEAM_REQUESTS, len(beam_outputs)))
+        delays = np.zeros((N_BEAM_REQUESTS, len(beam_outputs), n_ants, 2), np.float64)
         # Delay is in seconds, so needs to be very small
-        delays[..., 0] = rng.uniform(-1e-9, 1e-9, size=(len(beam_outputs), n_ants))
+        delays[..., 0] = rng.uniform(-1e-9, 1e-9, size=(N_BEAM_REQUESTS, len(beam_outputs), n_ants))
         # Phase is in radians
-        delays[..., 1] = rng.uniform(-2 * np.pi, 2 * np.pi, size=(len(beam_outputs), n_ants))
-
-        katcp_requests: dict[str, list[tuple[str, ...]]] = {
-            output.name: [
-                ("beam-weights", output.name, *weights[i]),
-                ("beam-quant-gains", output.name, quant_gains[i]),
-                ("beam-delays", output.name, *[f"{d[0]}:{d[1]}" for d in delays[i]]),
+        delays[..., 1] = rng.uniform(-2 * np.pi, 2 * np.pi, size=(N_BEAM_REQUESTS, len(beam_outputs), n_ants))
+
+        # NOTE: This list of lists is done in an order that has a set of requests for alternating
```

Comment:
"done" is normally associated with a verb, but "list of lists" is a noun. Do you mean
```suggestion
        # NOTE: This list of lists is constructed in an order that has a set of requests for alternating
```

Reply:
Quite right once again.